src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -403,9 +403,14 @@ def upload_stream(
             else:
                 raise ex
 
-    def notify_query_listeners(self, query_record: QueryRecord) -> None:
+    def notify_query_listeners(
+        self, query_record: QueryRecord, is_error: bool = False
+    ) -> None:
         with self._lock:
             for listener in self._query_listener:
+                # if listener is not set to record error query, skip
+                if is_error and not listener._record_error_queries:
+                    continue
```

Comment:
let's read this property similar to how we do it for `include_thread_id` and `include_describe`